gr-blocks/lib/nlog10_ff_impl.cc

```diff
@@ -51,11 +57,18 @@ namespace gr {
       const float *in = (const float *) input_items[0];
       float *out = (float *) output_items[0];
       int noi = noutput_items * d_vlen;
-      float n = d_n;
-      float k = d_k;
 
-      for (int i = 0; i < noi; i++)
-	out[i] = n * log10(std::max(in[i], (float) 1e-18)) + k;
+      for (int i = 0; i < noi; i++) {
+          out[i] = in[i] + d_float_min;
+      }
+
+      volk_32f_log2_32f(out, out, noi);
+      volk_32f_s32f_multiply_32f(out, out, d_n_log2_10, noi);
+      if (d_k != 0.0f) {
+        for (int i = 0; i < noi; i++) {
+          out[i] += d_k;
```

Comment:
Idea: instead of adding `d_k` here, we could just multiply the input with 2<sup>d_k</sup> and check whether it's truly > 0. Multiplication of floats is actually not any slower or less accurate than addition (quite the contrary, actually).

Reply:
Bonus: we have a VOLK multiply const kernel, `volk_32f_s32f_multiply_32f` :)

Reply:
(this is just an idea on how to speed things up further, and bears absolutely no criteria for acceptance of this PR! )